src/backend/cdb/cdbpartition.c

```diff
@@ -5228,7 +5228,18 @@ get_part_rule(Relation rel,
 
 		lc = list_head(l1);
 		prule2 = (PgPartRule *) lfirst(lc);
-		if (prule2 && prule2->topRule && prule2->topRule->children)
+
+		if (!prule2)
```

Comment:
I think lfirst cannot return NULL.

Reply:
This guards from `cell->data.ptr_value` that was explicitly initialized as `NULL`.

I found an example:
https://github.com/arenadata/gpdb/blob/03e22fbb7e91111e3bff7ceccaba5f7fc04c4d39/src/backend/executor/nodeMotion.c#L539

Reply:
I think this is a bad example. Why put a null pointer in the list?

Reply:
I'm think I'm fine with the current version. What do you think? Should this case be changed to an assert?

Reply:
may be